engine/src/main/java/org/terasology/rendering/dag/nodes/CopyToVRFrameBuffersNode.java

```diff
@@ -0,0 +1,121 @@
+import static org.lwjgl.opengl.GL11.GL_DEPTH_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.glClear;
+import static org.terasology.rendering.opengl.OpenGLUtils.renderFullscreenQuad;
+
+public class CopyToVRFrameBuffersNode extends AbstractNode {
+    private OpenVRProvider vrProvider;
+    private OpenVRStereoRenderer vrRenderer;
+
+    @In
+    private WorldRenderer worldRenderer;
+
+    @In
+    private Config config;
+
+    @In
+    private ScreenGrabber screenGrabber;
+
+    @In
+    private DisplayResolutionDependentFBOs displayResolutionDependentFBOs;
+
+    private RenderingDebugConfig renderingDebugConfig;
+    private RenderingConfig renderingConfig;
+    private Material finalPost;
+    private Material debug;
+
+    public void setOpenVRProvider(OpenVRProvider providerToSet) {
+        this.vrProvider = providerToSet;
+    }
+
+
+    @Override
+    public void initialise() {
+        renderingConfig = config.getRendering();
+        renderingDebugConfig = renderingConfig.getDebug();
+
+        finalPost = worldRenderer.getMaterial("engine:prog.post");
+        debug = worldRenderer.getMaterial("engine:prog.debug");
+        addDesiredStateChange(new BindFBO(FINAL));
+    }
+
+    @Override
+    public void process() {
+        PerformanceMonitor.startActivity("rendering/copyToVRFrameBuffers");
+        if (!renderingDebugConfig.isEnabled()) {
+            finalPost.enable();
+        } else {
+            debug.enable();
+        }
+
+        if (!renderingConfig.isVrSupport()) {
+            logger.warn("CopyToVRFrameBufferNode processed in non-VR mode.");
+        } else {
+            renderFinalStereoImage(worldRenderer.getCurrentRenderStage());
+        }
+
+        PerformanceMonitor.endActivity();
+    }
+
+    private void renderFinalStereoImage(RenderingStage renderingStage) {
+        if (this.vrRenderer == null) {
+            this.vrProvider.init();
+            this.vrRenderer = new OpenVRStereoRenderer(this.vrProvider, 1280, 720);
+        }
+        switch (renderingStage) {
+            case LEFT_EYE:
+                glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+                vrProvider.updateState();
+                org.lwjgl.opengl.EXTFramebufferObject.glBindFramebufferEXT(
+                        org.lwjgl.opengl.EXTFramebufferObject.GL_FRAMEBUFFER_EXT, vrRenderer.getTextureHandleForEyeFramebuffer(0));
+                glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+                renderFullscreenQuad(0, 0, 1280, 720);
+
+                break;
+
+            case RIGHT_EYE:
+                glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+                org.lwjgl.opengl.EXTFramebufferObject.glBindFramebufferEXT(
+                        org.lwjgl.opengl.EXTFramebufferObject.GL_FRAMEBUFFER_EXT, vrRenderer.getTextureHandleForEyeFramebuffer(1));
+                renderFullscreenQuad(0, 0, 1280, 720);
+                vrProvider.submitFrame();
+                GL11.glFinish();
```

Comment:
Refer to this example once again: https://github.com/ValveSoftware/openvr/blob/master/samples/hellovr_opengl/hellovr_opengl_main.cpp

Without `glFinish()` after `submitFrame()` (and also `glClear()` in some cases), the headset can flicker and act unpredictably. The good news is that, once I get it right on one computer, it seems to work on all other computers (that I have access to), so it's not a platform-specific thing.

If you see something strange with '`glClear()` or `glFinish()` in my commits down the road, it could be that I just be experimentally the thing that ended up working the best.